x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -683,4 +696,139 @@ private XContentBuilder getTokenIndexMappings() {
+                builder.startObject("properties");
+                {
+                    builder.startObject("uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("redirect_uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("enabled");
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject("user");
+                    {
+                        builder.field("type", "object");
+                        builder.startObject("properties");
+                        {
+                            builder.startObject("username");
+                            builder.field("type", "keyword");
+                            builder.endObject();
+
+                            builder.startObject("realm");
+                            {
+                                builder.field("type", "object");
+                                builder.startObject("properties");
+                                {
+                                    builder.startObject("name");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("type");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+
+                                    builder.startObject("domain");
+                                    builder.field("type", "keyword");
+                                    builder.endObject();
+                                }
+                                builder.endObject();
+                            }
+                            builder.endObject();
+
+                            builder.startObject("email");
+                            builder.field("type", "text");
+                            builder.endObject();
+
+                            builder.startObject("full_name");
+                            builder.field("type", "text");
+                            builder.endObject();
+
+                            builder.startObject("display_name");
+                            builder.field("type", "text");
+                            builder.endObject();
+                        }
+                        builder.endObject();
+                    }
+                    builder.endObject();
+
+                    builder.startObject("last_synchronized");
+                    builder.field("type", "date");
+                    builder.field("format", "epoch_millis");
+                    builder.endObject();
+
+                    builder.startObject("access");
+                    {
+                        builder.field("type", "object");
```

Comment:
![image](https://user-images.githubusercontent.com/4568420/145088577-acbde78f-de38-40c8-8d08-df8ea4fb95c5.png)

Reply:
Since we decided to not let past logins to affect user roles and privileges (search related), I think we should treat these past login event data as any other application specific data. Concretely, this means they are stored under the `application_data` field. Applications (Kibana in this case) can retrieve them and give them the meanings when presenting to users. But from ES's perspective, they are just another set of data. Since we don't provide search for them, we don't need to handle them specially.

Reply:
SGTM.